packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -63,24 +62,30 @@ function SelectToZoom(props: Props) {
       {...interactionProps}
     >
       {({ startPoint, endPoint }) => {
+        const [newStartPoint, newEndPoint] = transform(
+          startPoint,
+          endPoint,
+          xVisibleDomain,
+          yVisibleDomain
+        );
         return (
           <>
             <SelectionRect
-              startPoint={startPoint}
-              endPoint={endPoint}
+              startPoint={newStartPoint}
+              endPoint={newEndPoint}
               fill="white"
               stroke="black"
-              fillOpacity={keepRatio ? 0 : 0.25}
-              strokeDasharray={keepRatio ? '4' : undefined}
+              fillOpacity={0.25}
+              clampToVis
             />
+            {/* Draw un-transformed user-drawn rectangle */}
             {keepRatio && (
-              <RatioSelectionRect
+              <SelectionRect
```

Comment:
The diff can be a bit hard to process here.

1. The logic was inverted so that the actual zoom rectangle is always displayed. The `keepRatio` conditional is used to display the user-drawn rectangle instead. This allows to simplify the `fillOpacity` and `strokeDasharray` props.
1. The `RatioSelectionRect` was replaced by a regular `SelectionRect` that takes the transformed points with the addition of a prop `clampToVis` to keep the behaviour that was only in `RatioSelectionRect`. If `keepRatio` is `false`, the vis takes the full dims so it does not hurt to keep `clampToVis` to `true`.